### PR TITLE
IBX-8258: richtext: Html5Input is not washed for duplicate IDs

### PR DESCRIPTION
**Pull request by vidarl**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [IBX-8258](https://jira.ez.no/browse/IBX-8258)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | 3.3
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

XSLT stylesheet doesn't check if ids are unique. I wasn't able to reproduce the problem reported in [IBX-8258](https://jira.ez.no/browse/IBX-8258) with the alloy editor ( IDs are note included in the cut&paste operation when performed on Alloy it seems). It is reproducable on the CKEditor in 4.6
The problem is nevertheless represented in the XSLT stylesheet in 3.3 as well

Note: When merging to 4.6, remember to adopt for changes in XML namespaces in new fixtures

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: src/lib/eZ/RichText/Resources/stylesheets/xhtml5/edit/docbook.xsl [hunk 1]
```diff
@@ -85,13 +88,30 @@
     </xsl:choose>
   </xsl:template>
 
+  <xsl:template name="conditionallyAddIdAttribute">
```
Comment (Steveb-p) [inline src/lib/eZ/RichText/Resources/stylesheets/xhtml5/edit/docbook.xsl]: It's not actually conditionally adding ID attribute. It's changing the ID if a second one exists. We probably need a different name.

**Review by Steveb-p — APPROVED** [review]

File: tests/lib/eZ/RichText/Converter/Xslt/_fixtures/xhtml5/edit/lossy/010-duplicatedIds.docbook.xml [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    <para>Some <ezembedinline view="embed-inline-custom" ezxhtml:align="left" ezxhtml:class="embedClass" xlink:href="ezcontent://601" xml:id="id3_idm103"><ezlink ezxhtml:class="linkClass" xlink:href="ezcontent://106" xlink:show="new" xlink:title="Link title" xml:id="ezlinkembeddup_idm104"/></ezembedinline></para>
+    <para>Some <ezembedinline view="embed-inline-custom" ezxhtml:align="left" ezxhtml:class="embedClass" xlink:href="ezcontent://601" xml:id="id3_idm106"><ezlink ezxhtml:class="linkClass" xlink:href="ezcontent://106" xlink:show="new" xlink:title="Link title" xml:id="ezlinkembeddup_idm107"/></ezembedinline></para>
+    <eztemplate name="factoidbox" ezxhtml:align="center" ezxhtml:class="templateClass4" xml:id="ipsum"/>
+    <eztemplate name="factoidbox" ezxhtml:align="center" ezxhtml:class="templateClass4" xml:id="template"/>
+    <eztemplate name="factoidbox" ezxhtml:align="center" ezxhtml:class="templateClass4" xml:id="templatedup_idm110"/>
+    <eztemplate name="factoidbox" ezxhtml:align="center" ezxhtml:class="templateClass4" xml:id="templatedup_idm111"/>
+    <para>Some <eztemplateinline name="emojicake"/> for the otherwise unremarkable paragraph.</para>
+</section>
```
Comment (konradoboza) [inline tests/lib/eZ/RichText/Converter/Xslt/_fixtures/xhtml5/edit/lossy/010-duplicatedIds.docbook.xml]:
```suggestion
</section>

```

File: src/lib/eZ/RichText/Resources/stylesheets/xhtml5/edit/docbook.xsl [hunk 3]
```diff
@@ -9,6 +9,9 @@
     version="1.0">
   <xsl:output indent="yes" encoding="UTF-8"/>
 
+  <xsl:key name="ids" match="*[@id]" use="@id"/>
+  <xsl:key name="ids" match="ezxhtml5:a[@name]" use="@name"/>
```
Comment (konradoboza) [inline src/lib/eZ/RichText/Resources/stylesheets/xhtml5/edit/docbook.xsl:13]: Why both conditions are matched by the same `name="ids"`? Just asking, I am not familiar with such notation but maybe it does the job just fine.

Comment (vidarl) [reply src/lib/eZ/RichText/Resources/stylesheets/xhtml5/edit/docbook.xsl:13]: We are scanning for IDs using two patters and putting them in the same map (which is named `ids`). 

So we have html5 edit code like this:
`<p><a name="bottom"/>another anchor with the same name attribute.</p>`
where the value of the `name` attribute will also end up as an id, see [[xhtml5/edit/lossy/008-anchor.docbook.xml test](https://github.com/ezsystems/ezplatform-richtext/pull/243/files#diff-5dcffba4a6512771e0cca2bf0b7d65bfea0df58a748c3be6a1e92b9c0642ebc1)

File: src/lib/eZ/RichText/Resources/stylesheets/xhtml5/edit/docbook.xsl (hunk 1 above)
Comment (vidarl) [reply src/lib/eZ/RichText/Resources/stylesheets/xhtml5/edit/docbook.xsl]: @Steveb-p 
Renamed it to `addUniqueIdAttribute` in 104897d19dcfba421f178396f8d3088f108538a7.
ok ?

File: src/lib/eZ/RichText/Resources/stylesheets/xhtml5/edit/docbook.xsl (hunk 3 above)
Comment (konradoboza) [reply src/lib/eZ/RichText/Resources/stylesheets/xhtml5/edit/docbook.xsl:13]: Thank you, that makes sense.

**Review by konradoboza — APPROVED** [review]
Apart from nitpick https://github.com/ezsystems/ezplatform-richtext/pull/243/files#r1637781380.

**Review by Steveb-p — APPROVED** [review]

**Review by bogusez — APPROVED** [review]

**Pull request merged**